spec/lib/arbitration/redis_spec.rb

```diff
@@ -60,4 +60,68 @@
       end
     end
   end
+
+  context 'redis client connection params' do
+    context 'when only url is present' do
+      let(:redis_url) { "redis://redis.example.com:8888" }
+      let(:mailbox) {
+        MailRoom::Mailbox.new(
+          arbitration_options: {
+            redis_url: redis_url
+          }
+        )
+      }
+
+      it 'client has same specified url' do
+        subject.deliver?(123)
+
+        expect(redis.options[:url]).to eq redis_url
+      end
+
+      it 'client is a instance of Redis class' do
+        expect(redis).to be_a Redis
+      end
+    end
+
+    context 'when namespace is present' do
+      let(:namespace) { 'mail_room' }
+      let(:mailbox) {
+        MailRoom::Mailbox.new(
+          arbitration_options: {
+            namespace: namespace
+          }
+        )
+      }
+
+      it 'client has same specified namespace' do
+        expect(redis.namespace).to eq(namespace)
+      end
+
+      it 'client is a instance of RedisNamespace class' do
+        expect(redis).to be_a ::Redis::Namespace
+      end
+    end
+
+    context 'when sentinel is present' do
+      let(:redis_url) { 'redis://:mypassword@sentinel-master:6379' }
+      let(:sentinels) { [host: '10.0.0.1', port: '26379'] }
+      let(:mailbox) {
+        MailRoom::Mailbox.new(
+          arbitration_options: {
+            redis_url: redis_url,
+            sentinels: sentinels
+          }
+        )
+      }
+
+      before { ::Redis::Client::Connector::Sentinel.any_instance.stubs(:resolve).returns(sentinels) }
```

Comment:
It's really hard to make any kind of test/mock for sentinel... I've opted here for checking library API contract, but I'm open to any idea you can have.